src/network/p2p_topology.rs

```diff
@@ -1,22 +1,118 @@
 //! module defining the p2p topology management objects
 //!
 
+use bincode;
+use chain_core::property;
+use network_core::gossip;
 use poldercast::topology::{Cyclon, Module, Rings, Topology, Vicinity};
 use poldercast::Subscription;
 pub use poldercast::{Address, Id, InterestLevel, Node};
-use std::collections::BTreeMap;
-
-use std::sync::{Arc, RwLock};
+use std::{
+    collections::BTreeMap,
+    error, fmt,
+    io,
+    sync::{Arc, RwLock},
+    vec,
+};
 
 pub const NEW_TRANSACTIONS_TOPIC: u32 = 0u32;
 pub const NEW_BLOCKS_TOPIC: u32 = 1u32;
 
+#[derive(Debug, Clone)]
+pub struct Gossip(pub Vec<Node>);
+
+#[derive(Debug)]
+pub struct Error {
+    source: Box<dyn error::Error + Send + Sync>,
+}
+
+impl Error {
+    pub fn new<E>(source: E) -> Self
+    where
+        E: Into<Box<dyn error::Error + Send + Sync>>,
+    {
+        Error {
+            source: source.into(),
+        }
+    }
+}
+
+impl error::Error for Error {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        Some(self.source.as_ref())
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.source.fmt(f)
+    }
+}
+
+impl From<io::Error> for Error {
+    fn from(e:  io::Error) -> Self {
+        Error {
+            source: Box::new(e),
+        }
+    }
+}
+
+pub fn from_node_id(node_id: &gossip::NodeId) -> Id {
```

Comment:
Please remind me, why do we need these conversions rather than make `NodeId` an abstract associated type of `Gossip` and instantiating it here with `poldercast::Id`?

Reply:
the problem with keeping this abstract is that then we will need to abstract all the commands inside gossip message handler so we can work with that.

The other solution is to change rust-cardano Gossip trait and make NodeId generic there.
But I think that can be done in the separate commit if needed.